### DISCUSSION TITLE
Debian 12 "Bookworm" support

### DISCUSSION THREAD

**Discussion by kesnar** [regolith-linux/regolith-desktop#859] [Q&A] [answered]
Hello all,

Thank you for the great effort on creating and maintaining regolith!

I would like to ask you if there is currently official support for new Debian stable release (12 - Bookworm).

The following stopped working for me:
`deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/release-debian-bullseye-amd64 bullseye main`

And changing bullseye to bookworm, didn't return anything.

I have gathered from various posts here and there that there is a repo for testing release:
`deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/unstable-debian-testing-amd64 testing main`

But it is not clear to me if that works for debian testing (which would fast diverge out of debian 12).

Should I continue using this repo for the time being? Do you plan to migrate to debian 12?

(Also any plans on moving to the official debian repositories?)

Thank you. :)

**Comment by kgilmer** [comment]
Hi @kesnar 

> I would like to ask you if there is currently official support for new Debian stable release (12 - Bookworm).

The most recent production release of Regolith, 2.2, does not support Debian Bookworm.  The next release, 3.0, is planned to support it.  Build changes are in progress to implement this explicit support for Bookworm in addition to `testing`.

> The following stopped working for me:

Hmm, it's not expected that you would notice any change to releases at this point.  I may have broken something during my build updates, can you provide more details around "stopped working"?

> But it is not clear to me if that works for debian testing (which would fast diverge out of debian 12).

Right, things are in pre-release flux at the moment.  You'll be ok consuming from `testing` in the short term as nothing for a post-Bookworm release has been added, however once 3.0 is released it is strongly recommended to change your apt configuration to consume from the `bookworm` repo explicitly.  

The Regolith Desktop landing page will be updated with installation instructions for Debian Bookworm once the release is complete. 

> (Also any plans on moving to the official debian repositories?)

I've had some brief discussions with core Debian contributors on the subject and would like to put energy into this when time allows, and also  would welcome any contributions.

**Reply by kesnar** [reply]
Hello @kgilmer,

Thanks for the answer.

Unfortunately I didn't keep the logs from the update, neither can I find something specific in my `/var/log/apt/` directory. But the story more or less is the following:

I was in Debian 11. To upgrade to Debian 12, I removed any "unofficial" sources and did a `apt full-upgrade`, with updated sources. I re-entered the "unofficial" source (including regolith) and did a second `apt full-upgrade`. Regolith was complaining that some dependency could not be met. I don't remember the exact package, but I think it was something along the lines of "gnome control center", or something like that. But unfortunately I cannot confirm. I searched around the internet and stumbled on a post made by you, about the `testing`  branch, tried and it worked.

Bear in mind that I did the upgrade two days before the official release, but I don't think that much has changed since.

I would love to help (when time allows). I don't have any experience with repo maintenance, but I will learn. If you have some way I follow the project and help, please do share. Thank you!

**Comment by kgilmer** [comment, marked as answer]
Regolith 3.0 beta 1 has been released, info here: https://regolith-desktop.com/docs/reference/Releases/regolith-3.0-release-notes/
